Update circle.py

empirical-dan opened the pull request:
Added attributes x0, y0 (co-ordinates of centre of circle) to Circle Class object instance. A Circle object is defined by the centre (x0, y0) in the constructor. However, there were no attributes corresponding to this only x, y which are x0-r, y0-r due to inheritance from RoundRect (most shapes are defined by the top-left corner co-ordinate x, y). The addition of the centre of Circle attributes make it easier to locate the circle on the screen using the same reference co-ordinates used to define it.

empirical-dan committed:
Update circle.py

Added attributes x0, y0 (co-ordinates of centre of circle) to Circle Class object instance. A Circle object is defined by the centre (x0, y0) in the constructor. However, there were no attributes corresponding to this only x, y which are x0-r, y0-r due to inheritance from RoundRect (most shapes are defined by the top-left corner co-ordinate x, y). The addition of the centre of Circle attributes make it easier to locate the circle on the screen using the same reference co-ordinates used to define it.

kmatch98 commented:
Cool, it's your first pull request.  I know it can be daunting, I'm relatively new at using git and github myself over the past year or so, and some things (like formatting requirements) can be really annoying.  There are some tools to make it easier and you'll be able to figure it out.  

-------

Before we get too deep into the weeds, I'll make a couple of comments on your code.  First, when adding these two variables it will add to the memory overhead to each circle object.  It probably won't be a big deal to store a couple of integers, but on microcontrollers we always want to be conscious about any extra memory use.

1) Reduce the number of variables you need to store to just the radius.  Store the radius and then calculate what you want based on the self.x and self.y values.  Change the stored variable to `self.r` and then create two getters for `x0` and `y0`.  

You said you are new to Python, so go look at [how the getter for the fill color is created in RoundRect](https://github.com/adafruit/Adafruit_CircuitPython_Display_Shapes/blob/5d8bffe99b4d9d089f3c495d6cd9146bef54e927/adafruit_display_shapes/roundrect.py#L144-L148) using the `@property` decorator.

If you're ambitious, you can also create a two new setters, one each `x0` and `y0`.

Oh, and please remove the comment with your name in it.  You'll be immortalized forever through this pull request, folks can follow this paper trail to see your contribution.  You could imagine what the code would look like if there were [comment fingerprints of everyone that touched the code and all the changes.  You can see there's already been 123 commits to this relatively modest library.](https://github.com/adafruit/Adafruit_CircuitPython_Display_Shapes/commits/master)

Update your code and then I can help  you through the gauntlet of the code checks.

--------

We'll eventually need to get your code to pass the code checks so you get a "green check", and I can help you get there.  

- Here's a [great learning guide about running pre-commit to help you clear all the code checks](https://learn.adafruit.com/creating-and-sharing-a-circuitpython-library/check-your-code) for detailed instructions.
- I can walk you through it, or I can run through the checks and make the changes and then you can merge it into your PR.

empirical-dan committed:
Update circle.py

empirical-dan commented:
Hi
I've added just the radius attribute (r) as suggested and implemented getters using @property so that x0 and y0 can be accessed as if they were attributes but saving memory on the MCU.
Could you possibly explain further in what context we could use setters for the centre of the circle individually (i.e. an individual setter for each of x0 and y0)? I thought that the user would need to draw a new circle? Do you mean that the getters should be able to move the existing circle to a new location?
Best
Dan

kmatch98 commented:
Yep add setters for resetting the x0 y0 referenced to the center in case you don’t want to recreate the object but just move it in reference to the center. 

So those setters you’ll update self.x and self.y based on the input x0 or y0 values. (To be clear about it, the self.x and .y come from a parent class).

empirical-dan commented:
I have added further code for the setters in a separate request.
